src/resources/Resources.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useMemo, useRef, useEffect } from "react";
+import styles from './Resources.module.scss';
+import { updateBackend } from './helpers/actions';
+import { sortResources, startScrolling, stopScrolling } from './helpers/utils';
+
+export default function Resources({ available_resources, relative_url_root }) {
+    const sortedResources = useMemo(() => sortResources(available_resources), [available_resources]);
+    const [resources, setResources] = useState(sortedResources);
+    const draggedIndexRef = useRef(null);
+    const scrollIntervalRef = useRef(null);
+
+    useEffect(() => {
+        setResources(sortedResources);
+    }, [sortedResources]);
+
+    const handleDragStart = (e, index) => {
+        draggedIndexRef.current = index;
+    };
+
+    const handleDragOver = (e, index) => {
+        e.preventDefault();
+        if (draggedIndexRef.current === index) return;
+
+        const newResources = [...resources];
+        const draggedItem = newResources[draggedIndexRef.current];
+        newResources.splice(draggedIndexRef.current, 1);
+        newResources.splice(index, 0, draggedItem);
+        
+        draggedIndexRef.current = index;
+        setResources(newResources);
+
+        // Scroll logic to start scroll scroll when dragging items
+        const { clientY } = e;
+        const scrollThreshold = 130;
+
+        if (clientY < scrollThreshold) {
+            startScrolling(-1, scrollIntervalRef);
+        } else if (window.innerHeight - clientY < scrollThreshold) {
+            startScrolling(1, scrollIntervalRef);
+        } else {
+            stopScrolling(scrollIntervalRef);
+        }
+    };
+
+    const handleDrop = () => {
+        draggedIndexRef.current = null;
+        stopScrolling(scrollIntervalRef);
+        updateBackend(relative_url_root, resources);
+    };
```

Comment:
If it's not too much work, it might be nice to break the drag-to-reorder logic out into a separate component. I can imagine it's something we might want to use elsewhere.